Sources/Accord.Statistics/Filters/Codification`1.Options.cs

```diff
@@ -536,10 +536,22 @@ private void TryAddValue(object value)
                 {
                     T key = (T)value;
                     if (!Mapping.ContainsKey(key))
-                        Mapping.Add(key, Mapping.Count); // And register the String->Integer mapping
+                        Mapping.Add(key, FindNextAvailableValue(Mapping)); // And register the String->Integer mapping
                 }
             }
 
+            private int FindNextAvailableValue(TwoWayDictionary<T, int> mapping)
+            {
+                // Start checking from 0 upwards for the next available value
+                int nextValue = 0;
+                while (mapping.Values.Contains(nextValue))

```

Comment:
Performance verschlechterung von O(1) -> O(n²)

Ideen um Performance im Rahmen zu halten:

Pack die ICollection in ein `HashSet<int> numSet` dann kannst du `while(numSet.Contains(nextValue)){++nextValue;}` machen, dadurch ist es wenigstens O(n).
FunFact: Der Hash bei einem HashSet<int> ist die Zahl selber xD
 
Was auch gehen würde wäre die Collection zu sortieren und dann einmal drüber zu iterieren um eine Lücke in den Zahlen zu finden, wäre wegen dem sortieren aber O(n log n)